src/main/java/io/gravitee/policy/retry/RetryPolicy.java

```diff
@@ -0,0 +1,260 @@
+                        connection.sendResponse();
+                    } else {
+                        DirectProxyConnection connection = new DirectProxyConnection(HttpStatusCode.BAD_GATEWAY_502);
+                        handler.handle(connection);
+                        connection.sendResponse();
+                    }
+                }
+            });
+        }
+    }
+
+    public static class RetryProxyConnection implements ProxyConnection {
+
+        private final ProxyConnection wrapped;
+        private final ProxyResponse response;
+        private Handler<ProxyResponse> responseHandler;
+
+        RetryProxyConnection(ProxyConnection connection, ProxyResponse response) {
+            this.wrapped = connection;
+            this.response = response;
+        }
+
+        @Override
+        public ProxyConnection cancel() {
+            return wrapped.cancel();
+        }
+
+        @Override
+        public ProxyConnection exceptionHandler(Handler<Throwable> exceptionHandler) {
+            return wrapped.exceptionHandler(exceptionHandler);
+        }
+
+        @Override
+        public ProxyConnection responseHandler(Handler<ProxyResponse> responseHandler) {
+            this.responseHandler = responseHandler;
+            return this;
+        }
+
+        @Override
+        public WriteStream<Buffer> write(Buffer buffer) {
+            return wrapped.write(buffer);
+        }
+
+        @Override
+        public void end() {
+            wrapped.end();
+        }
+
+        @Override
+        public void end(Buffer buffer) {
+            wrapped.end(buffer);
+        }
+
+        @Override
+        public WriteStream<Buffer> drainHandler(Handler<Void> drainHandler) {
+            return wrapped.drainHandler(drainHandler);
+        }
+
+        @Override
+        public boolean writeQueueFull() {
+            return wrapped.writeQueueFull();
+        }
+
+        public void sendResponse() {
+            this.responseHandler.handle(response);
+        }
+    }
+
+    public static class DirectProxyConnection implements ProxyConnection {
+
+        private Handler<ProxyResponse> responseHandler;
+
+        private final ProxyResponse response;
+
+        public DirectProxyConnection(int statusCode) {
+            this.response = new EmptyProxyResponse(statusCode);
+        }
+
+        @Override
+        public WriteStream<Buffer> write(Buffer content) {
+            throw new IllegalStateException();
+        }
+
+        @Override
+        public void end() {
+            // Nothing to do here...
+        }
+
+        @Override
+        public ProxyConnection responseHandler(Handler<ProxyResponse> responseHandler) {
+            this.responseHandler = responseHandler;
+            return this;
+        }
+
+        public void sendResponse() {
+            this.responseHandler.handle(response);
+        }
+    }
+
+    public static class EmptyProxyResponse implements ProxyResponse {
```

Comment:
This class exists in `gravitee-gateway-core`, would it be interesting to share it via `gravitee-gateway-api` ?